### PR TITLE
Consolidate kubernetes-dashboard into kubernetes application

### PR DESCRIPTION
**Pull request by rodrigodelmonte**
This PR consolidates the kubernetes-dashboard deployment to the kubernetes application, and bumps the container versions.

### CODE REVIEW

File: cluster/manifests/dashboard/deployment.yaml [hunk 1]
```diff
@@ -31,7 +31,7 @@ spec:
       serviceAccountName: kubernetes-dashboard
       containers:
       - name: kubernetes-dashboard
-        image: registry.opensource.zalan.do/teapot/kubernetes-dashboard:v2.0.4
+        image: registry.opensource.zalan.do/teapot/kubernetes-dashboard:v2.4.0
```
Comment (rodrigodelmonte) [inline cluster/manifests/dashboard/deployment.yaml:34]: https://github.bus.zalan.do/teapot/kubernetes-dashboard-pipeline/pull/19

**Comment by rodrigodelmonte** [comment]
:+1:

File: cluster/manifests/dashboard/deployment.yaml [hunk 2]
```diff
@@ -4,22 +4,22 @@ metadata:
   name: kubernetes-dashboard
   namespace: kube-system
   labels:
-    application: kubernetes-dashboard
-    component: dashboard
-    version: v2.0.4
+    application: kubernetes
+    component: kubernetes-dashboard
```
Comment (mikkeloscar) [inline cluster/manifests/dashboard/deployment.yaml]: nit: maybe the component name should just be `dashboard`?

File: cluster/manifests/dashboard/scraper.yaml [hunk 3]
```diff
@@ -23,28 +23,28 @@ metadata:
...
+    component: dashboard-metrics-scraper
+    version: v1.0.7
 spec:
   replicas: 1
   selector:
     matchLabels:
-      application: kubernetes-dashboard
-      component: metrics-scraper
+      application: kubernetes
+      component: dashboard-metrics-scraper
```
Comment (mikkeloscar) [inline cluster/manifests/dashboard/scraper.yaml]: let's use `deplotyment: dashboard-metrics-scraper` as the `matchLabels` like we did for the other consolidated components.

File: cluster/manifests/deletions.yaml [hunk 4]
```diff
@@ -31,6 +31,14 @@ pre_apply:
     application: stackset-controller
   namespace: kube-system
   kind: Deployment
+- labels:
+    application: kubernetes-dashboard
+  namespace: kube-system
+  kind: Deployment
+  - labels:
```
Comment (mikkeloscar) [inline cluster/manifests/deletions.yaml]: This is indented too much which makes CLM fail in the e2e create cluster step.

Comment (rodrigodelmonte) [reply cluster/manifests/deletions.yaml]: Wops, thanks!

**Comment by rodrigodelmonte** [comment]
:+1:

**Comment by myaser** [comment]
👍🏻

**Pull request merged**